### PR TITLE
ref(app-store-connect): Add adjusts

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Removes not used App Store Connect Context file (app/views/settings/project/appStoreConnectContext.tsx)
- [x] Updates Global Alert to support multiple App Store Connect instances 
- [x] Updates Custom Repositories code to support multiple App Store Connect instances 
- [x] Moves types/functions around
- [x] Temp removed Alert Dismiss button

**Before:**

Screenshot 2021-11-08 at 20 57 56

**After:**

image

### CODE REVIEW

**Comment by relaxolotl** [comment]
i regret what i said yesterday, adding in prompts that work the way app store connect needs them requires a pretty cursed implementation: https://github.com/getsentry/sentry/pull/29870

@priscilawebdev it may be a good idea to just continue with dismiss disabled and we just defer adding support for that until later on in the quarter; i'll need to pick at the brains of those who are involved in prompts because as far as i can tell this use case isn't supported at all by prompts. this means we're going to have to extend prompts in a way that's unique to this current use case.

i've also rebased and pushed some minor fixes, one for a regression and one to update endpoints to reflect changes that were made to the new `/status/` endpoint.

File: static/app/views/settings/project/projectSettingsNavigation.tsx [hunk 1]
```diff
@@ -14,7 +15,10 @@ type Props = {
 const ProjectSettingsNavigation = ({organization, project}: Props) => {
   const appStoreConnectContext = useContext(AppStoreConnectContext);
 
-  const debugFilesNeedsReview = !!appStoreConnectContext?.updateAlertMessage;
+  const debugFilesNeedsReview = some(
+    appStoreConnectContext,
+    srcStatus => srcStatus.credentials.status === 'invalid'
```
Comment (priscilawebdev) [inline static/app/views/settings/project/projectSettingsNavigation.tsx:20]: good catch!!

**Review by Swatinem — APPROVED** [review]

File: static/app/utils/appStoreValidationErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+export function getAppStoreValidationErrorMessage(
+  error: AppStoreConnectValidationError,
+  repo?: {name: string; link: string}
+) {
+  switch (error.code) {
+    case 'app-connect-authentication-error':
+      return repo
+        ? tct(
+            'Apple Store Connect credentials are invalid or missing. [linkToCustomRepository]',
```
Comment (matejminar) [inline static/app/utils/appStoreValidationErrorMessage.tsx:19]:
```suggestion
            'App Store Connect credentials are invalid or missing. [linkToCustomRepository]',
```

File: static/app/utils/appStoreValidationErrorMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    "Make sure the credentials of the '[customRepositoryName]' repository are correct and exist.",
+                    {
+                      customRepositoryName: repo.name,
+                    }
+                  )}
+                </Link>
+              ),
+            }
+          )
+        : t('The supplied Apple Store Connect credentials are invalid or missing.');
```
Comment (matejminar) [inline static/app/utils/appStoreValidationErrorMessage.tsx:33]:
```suggestion
        : t('The supplied App Store Connect credentials are invalid or missing.');
```

File: static/app/utils/appStoreValidationErrorMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    case 'app-connect-forbidden-error':
+      return t('The supplied API key does not have sufficient permissions.');
+    case 'app-connect-multiple-sources-error':
+      return t(
+        'Only one Apple App Store Connect application is allowed in this project.'
```
Comment (matejminar) [inline static/app/utils/appStoreValidationErrorMessage.tsx:38]:
```suggestion
        'Only one App Store Connect application is allowed in this project.'
```

**Review by matejminar — APPROVED** [review]

**Pull request merged**